lib/events/athena/consumer.go

```diff
@@ -129,12 +136,41 @@ func newConsumer(cfg Config) (*consumer, error) {
 			}
 			return fw, nil
 		},
+		cancelRun: cancelFn,
+		finished:  make(chan struct{}),
 	}, nil
 }
 
 // run continuously runs batching job. It is blocking operation.
 // It is stopped via canceling context.
 func (c *consumer) run(ctx context.Context) {
+	defer close(c.finished)
+	// TODO(tobiaszheller): consider adding feature flag if locking cause problems
+	// during testing on cloud. We could have switch to either run processEventsContinuously
+	// or runContinuouslyOnSingleAuth.
+	c.runContinuouslyOnSingleAuth(ctx, c.processEventsContinuously)
+}
+
+// Close is used to close consumer.run running in background.
+func (c *consumer) Close() error {
+	if c.cancelRun != nil {
+		c.cancelRun()
+	}
+	defer c.logger.Debug("Consumer finished")
+	select {
+	case <-c.finished:
+		return nil
+	case <-time.After(1 * time.Second):
+		// ctx is use through all calls within consumer.Run so it should finished
+		// very fast, within miliseconds.
+		return errors.New("consumer not finished in time, returning earlier")
+	}
```

Comment:
if we do it asynchronously, lock won't be released, because backend is closed soon after. I have started initially with asynchronously and because of that I have added that synchronous shutdown.